### ISSUE TITLE
ParserServices.esTreeNodeToTSNodeMap.get returns any

### ISSUE THREAD

**Issue by ilyub** [typescript-eslint/typescript-eslint#4162]
**Repro**

```ts
export function f(parser: ParserServices, node: TSESTree.Node): void {
  const result = parser.esTreeNodeToTSNodeMap.get(node);
}
```

**Expected Result**

"result" should have typescript node type (e.g. TypeReferenceNode)

Works as expected with @typescript-eslint/experimental-utils@5.3.0

**Actual Result**

After upgrading to @typescript-eslint/experimental-utils@5.4.0
"result" has any type.

**Additional Info**

Had to downgrade to @typescript-eslint/experimental-utils@5.3.0

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/experimental-utils` | `5.4.0` |
| `TypeScript`                            | `4.4.4` |
| `node`                                  | `14` |

**Comment by bradzacher** [comment]
This is due to #4164
